### DISCUSSION TITLE
Can I copy code directly from the repository?

### DISCUSSION THREAD

**Discussion by ericoporto** [dotnet/msbuild#7399] [Q&A] [answered]
Hi, 

I would like to use [`CommandLineBuilder`](https://github.com/dotnet/msbuild/blob/d1f5a8af82872bafadb82227ec99f2fcd7b731ec/src/Utilities/CommandLineBuilder.cs) in an application since it appears to handle safe escaping of parameters nicely. But I really don't need anything else from MSBuild. 

If I just copy the MSBuild license somewhere in my repository, would it be ok to do this? Yes I will adapt the source to not depend on other things. I found escaping arguments really complex but I am currently in need of  invoking an external program from my C# code.

Unfortunately I could not find anything as good in the default library of .NET.

**Comment by rainersigwald** [comment, marked as answer]
There is a https://github.com/dotnet/msbuild/blob/main/LICENSE in the repo that describes what you may do with the code. Interpreting it is up to you or your lawyers and I won't offer you any legal advice on that front, since it might be interpreted as coming from my employer, Microsoft.

**Reply by ericoporto** [reply]
Thanks, I will assume the `modify` clause makes it alright to grab a small piece of it if I include said license.
